vllm/inputs/registry.py

```diff
@@ -235,29 +241,55 @@ def wrapper(model_cls: N) -> N:
 
         return wrapper
 
-    def process_input(self, model_config: "ModelConfig",
-                      inputs: LLMInputs) -> LLMInputs:
+    def _process_input(self, inputs: LLMInputs, model_config: "ModelConfig",
+                       processor: Callable, **processor_kwargs) -> LLMInputs:
         """
-        Apply an input processor to an instance of model inputs.
+        Apply an input processor to an instance of model inputs. This will
+        usually not be invoked be directly, and instead will be wrapped in
+        a functools partial once the processor is created.
 
         The model is identified by ``model_config``.
 
         See also:
             :ref:`input_processing_pipeline`
         """
+        return processor(InputContext(model_config), inputs,
+                         **processor_kwargs)
+
+    def create_input_processor(self, model_config: "ModelConfig") -> Callable:
+        """
+        Create an input processor (see :meth:`_process_input`) for a
+        specific model.
+        """
+        # Determine which kwargs can be leveraged for the input processor
+        # and drop + warn for kwargs that are unimplemented.
+        # NOTE: we don't allow override values for ctx/inputs, since doing
+        # so can lead to value collisions etc.
+        processor = self._get_model_input_processor(model_config)
+        processor_kwargs = get_allowed_kwarg_only_overrides(
+            callable=processor, overrides=model_config.processor_kwargs)
+        return functools.partial(self._process_input,
+                                 model_config=model_config,
+                                 processor=processor,
+                                 **processor_kwargs)
+
+    def _get_model_input_processor(self,
+                                   model_config: "ModelConfig") -> Callable:
```

Comment:
```suggestion
                                   model_config: "ModelConfig"):
```

Reply:
If you do want to explicitly specify it, it should return `InputProcessor` here.